crates/core/src/vm.rs

```diff
@@ -37,10 +39,14 @@ pub fn build_query<'a>(
         result = match op {
             Query::IndexScan(IndexScan {
                 table,
-                col_id,
+                columns,
                 lower_bound,
                 upper_bound,
-            }) if db_table => iter_by_col_range(stdb, tx, table, col_id, (lower_bound, upper_bound))?,
+            }) if db_table => {
+                assert_eq!(columns.len(), 1, "Only support single column IndexScan");
```

Comment:
Apologies if this has been covered before by Mazdak: are we ok with panicking in places like this that we don't yet support, instead of returning an Err?